include/libopencm3/stm32/fdcan.h

```diff
@@ -0,0 +1,1420 @@
+#define FDCAN_CCU_CWDR				MMIO32((CAN_CCU_BASE) + 0x000C)
+/** Clock calibration unit interrupt register (FDCAN_CCU_IR) */
+#define FDCAN_CCU_IRR				MMIO32((CAN_CCU_BASE) + 0x0010)
+/** Clock calibration unit interrupt enable register (FDCAN_CCU_IE) */
+#define FDCAN_CCU_IER				MMIO32((CAN_CCU_BASE) + 0x0014)
+/**@}*/
+
+/* -------------------------------------------------------------------------- */
+/** @ingroup fdcan_defines  @{ */
+
+/* --- FDCAN_CREL: (FDCAN_CREL) --------------------------------------------- */
+/* Core release */
+#define FDCAN_CREL_REL_MASK 0xF
+#define FDCAN_CREL_REL_SHIFT 28
+/* Step of core release */
+#define FDCAN_CREL_STEP_MASK 0xF
+#define FDCAN_CREL_STEP_SHIFT 24
+/* Sub-step of core release */
+#define FDCAN_CREL_SUBSTEP_MASK 0xF
+#define FDCAN_CREL_SUBSTEP_SHIFT 20
+/* Timestamp Year */
+#define FDCAN_CREL_YEAR_MASK 0xF
+#define FDCAN_CREL_YEAR_SHIFT 16
+/* Timestamp Month */
+#define FDCAN_CREL_MON_MASK 0xFF
+#define FDCAN_CREL_MON_SHIFT 8
+/* Timestamp Day */
+#define FDCAN_CREL_DAY_MASK 0xFF
+#define FDCAN_CREL_DAY_SHIFT 0
+
+/* --- FDCAN_ENDN: FDCAN Endian register ------------------------------------ */
+/* Endianness Test value */
+#define FDCAN_ENDN_ETV_MASK 0xFFFFFFFF
+#define FDCAN_ENDN_ETV_SHIFT 0
+
+/* --- FDCAN_DBTP: FDCAN data bit timing and prescaler register ------------- */
+/* Transceiver Delay Compensation */
+#define FDCAN_DBTP_TDC BIT23
+/* Data bitrate prescaler */
+#define FDCAN_DBTP_DBRP_MASK 0x1F
+#define FDCAN_DBTP_DBRP_SHIFT 16
+/* Data time segment before sample point */
+#define FDCAN_DBTP_DTSEG1_MASK 0x1F
+#define FDCAN_DBTP_DTSEG1_SHIFT 8
+/* Data time segment after sample point */
+#define FDCAN_DBTP_DTSEG2_MASK 0xF
+#define FDCAN_DBTP_DTSEG2_SHIFT 4
+/* Synchronization jump width */
+#define FDCAN_DBTP_DSJW_MASK 0xF
+#define FDCAN_DBTP_DSJW_SHIFT 0
+
+/* --- FDCAN_TEST: (FDCAN_TEST) --------------------------------------------- */
+/* Receive pin */
+#define FDCAN_TEST_RX BIT7
+/* Control of transmit pin */
+enum fdcan_test_tx {
+	FDCAN_TEST_TX_RESET		= 0,
+	FDCAN_TEST_TX_SAMPLE	= 1,
+	FDCAN_TEST_TX_DOMINANT	= 2,
+	FDCAN_TEST_TX_RECESSIVE	= 3,
+	FDCAN_TEST_TX_MASK		= 3,
+};
+#define FDCAN_TEST_TX_SHIFT 5
+/* Loop back mode */
+#define FDCAN_TEST_LBCK BIT4
+
+/* --- FDCAN_RWD: FDCAN RAM watchdog register ------------------------------- */
+/* Watchdog value */
+#define FDCAN_RWD_WDV_MASK 0xFF
+#define FDCAN_RWD_WDV_SHIFT 8
+/* Watchdog configuration */
+#define FDCAN_RWD_WDC_MASK 0xFF
+#define FDCAN_RWD_WDC_SHIFT 0
+
+/* --- FDCAN_CCCR: FDCAN CC control register -------------------------------- */
+/* Non ISO operation */
+#define FDCAN_CCCR_NISO BIT15
+/* Edge Filtering during bus Integration */
+#define FDCAN_CCCR_EFBI BIT13
+/* Protocol exception handling disable */
+#define FDCAN_CCCR_PXHD BIT12
+/* FDCAN bitrate Switching */
+#define FDCAN_CCCR_BRSE BIT9
+/* FD operation enable */
+#define FDCAN_CCCR_FDOE BIT8
+/* Test mode enable */
```

Comment:
If you add the second * to all of these, they get doxygen too :)  (no big deal, but when you have the descriptions already)